src/app/Http/Controllers/BusinessTripController.php

```diff
@@ -35,46 +45,133 @@ public function create() {
+
+        $message = '';
+        $recipient = 'admin@example.com';
+        $viewTemplate = 'emails.new_trip_admin';
+
+        // Create an instance of the SimpleMail class
+        $email = new SimpleMail($message, $recipient, $viewTemplate);
+
+        // Send the email
+        Mail::to($recipient)->send($email);
+
+        //Redirecting to the homepage
+        return redirect()->route('components.homepage');
 
     }
 
     /**
      * Returning the view with the trip editing form
      */
-    public function edit(BusinessTrip $b) {
-        return view();
+    public function edit(BusinessTrip $trip) {
+        return view('business-trips.edit', ['trip' => $trip]);
+
     }
 
     /**
      * Validation as in create(), should check if user is admin/not
      * Redirecting to the trip editing form
      * Sending mail with mail component to admin
      */
-    public function update(Request $request, BusinessTrip $b) {
+    public function update(Request $request, BusinessTrip $trip) {
+        //Validate data
+        $validatedData = $request->validate([
+            //Add check if user is admin/not
+            //Add more fields and validation rules
+            'place' => 'required|varchar|max:200',
+            'datetime_start' => 'required|datetime',
+        ]);
+
+        //Update the trip based on the validated data
+        $trip->update($validatedData);
+
+        //Sending mails
+
+        $message = '';
+        $recipient = 'admin@example.com';
+        $viewTemplate = 'emails.new_trip_admin';
+
+        // Create an instance of the SimpleMail class
+        $email = new SimpleMail($message, $recipient, $viewTemplate);
+
+        // Send the email
+        Mail::to($recipient)->send($email);
+
+        return redirect()->route('business-trips.edit', $trip);
 
     }
 
     /**
-     * Next 3 functions could be combined to changeState(Request $request, BuisenessTrip $b)
+     * Next 3 functions could be combined to changeState(Request $request, BusinessTrip $b)
+     * Check for admin needed?
      *
      * Updating state of the trip to cancelled
      * Adding cancellation reason
      */
-    public function cancel(BusinessTrip $b) {
+    public function cancel(BusinessTrip $trip) {
+        //Cancel the trip and add cancellation reason
+        $trip->update(['state' => 'canceled', 'cancellation_reason' => request('cancellation_reason')]);
+
+        //Send cancellation email to admin
+        $message = '';
+        $recipient = 'admin@example.com';
+        $viewTemplate = 'emails.cancellation_request_admin';
+
+        // Create an instance of the SimpleMail class
+        $email = new SimpleMail($message, $recipient, $viewTemplate);
+
+        // Send the email
+        Mail::to($recipient)->send($email);
+
+        return redirect()->route('business-trips.show', $trip);
 
     }
 
     /**
      * Same as update(), updating state of the trip to confirmed
      */
-    public function confirm() {
```

Comment:
bolo by fajn k týmto metódam ktoré menia stav, prirobiť nejaký check na to, či sa volajú na ceste so správnym aktuálnym stavom, teda či sa nejde zmeniť na nový nezmyselne